[Lua] cmake find_package(lua REQUIRED) failure after vcpkg installation

Thanks for vcpkg. I find it is exactly what I need.

I am building software in a (relatively) barebones Ubuntu 16.04 Xenial docker container with vcpkg installed in folder` /vcpkg`. I can successfully install lua[cpp] along with several other C/C++ packages. But when I use the vcpkg manager within my own CMake project it fails to find the lua package.

**Host Environment**
- OS: [e.g. Ubuntu 16.04 Docker]
- Compiler: revision gcc 5.4.0, cmake 3.21.4

**To Reproduce**
Steps to reproduce the behavior:

Install lua[cpp]

`./vcpkg install lua[cpp]`

Then try to build a personal CMake project which uses lua, e.g.

```
find_package     (Boost REQUIRED COMPONENTS thread filesystem)                   # Find boost. 
find_package     (yaml-cpp CONFIG REQUIRED)                                                     # Find yaml-cpp. 
find_package     (cppnetlib CONFIG REQUIRED)                                                     # Find yaml-cpp. 
find_package     (lua REQUIRED)                                                                             # Find lua.
find_package     (ZeroMQ CONFIG REQUIRED)
```
using the cmake command that includes the vcpkg toolchain.  CMake (version 3.21.4)

```
# cmake -B build -S . -DCMAKE_TOOLCHAIN_FILE=/vcpkg/scripts/buildsystems/vcpkg.cmake
-- The C compiler identification is GNU 5.4.0
-- The CXX compiler identification is GNU 5.4.0
.
.
.
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE
-- Found Boost: /vcpkg/installed/x64-linux/include (found version "1.78.0") found components: thread filesystem chrono atomic
CMake Error at /vcpkg/installed/x64-linux/share/lua/vcpkg-cmake-wrapper.cmake:11 (_find_package):
  Could not find a package configuration file provided by
  "unofficial-lua-cpp" with any of the following names:

    unofficial-lua-cppConfig.cmake
    unofficial-lua-cpp-config.cmake

  Add the installation prefix of "unofficial-lua-cpp" to CMAKE_PREFIX_PATH or
  set "unofficial-lua-cpp_DIR" to a directory containing one of the above
  files.  If "unofficial-lua-cpp" provides a separate development package or
  SDK, be sure it has been installed.
Call Stack (most recent call first):
  /vcpkg/scripts/buildsystems/vcpkg.cmake:747 (include)
  CMakeLists.txt:114 (find_package)

```
**The issue seems to be that the file `unofficial-lua-cpp-config.cmake` has been installed into the wrong location by vcpkg.** This file is installed by vcpkg in folder `/vcpkg/packages/lua_x64-linux/share/share/unofficial-lua-cpp` .  I suspect the `share/share/` part of the path is incorrect and is probably a simple typo in someone's code.



**Failure logs** 
-Cut and paste the appropriate build messages from the console output.
-Please attach any additional failure logs mentioned in the console output.

**Additional context**
Add any other context about the problem here, such as what you have already tried to resolve the issue.

Running CMake with the incorrect folder explicitly specified as suggested by the error message rectifies the problem,

``` 
cmake -B build -S . -DCMAKE_TOOLCHAIN_FILE=/vcpkg/scripts/buildsystems/vcpkg.cmake 
-Dunofficial-lua-cpp_DIR=/vcpkg/packages/lua_x64-linux/share/share/unofficial-lua-cpp
```
and cmake successfully generates the build environment. So I have a fix to build my code but this is clearly not the intent of vcpkg which is trying to hide these details from end-users like me. I have attached a copy of the CMakeLists.txt I use for my code for your inspection. You will probably not be able to compile this file with cmake as it is building a custom project not available to you.

[CMakeError.log](https://github.com/microsoft/vcpkg/files/8147072/CMakeError.log)
[CMakeOutput.log](https://github.com/microsoft/vcpkg/files/81
[console.txt](https://github.com/microsoft/vcpkg/files/8147075/console.txt)
47073/CMakeOutput.log)
[CMakeLists.txt](https://github.com/microsoft/vcpkg/files/8147077/CMakeLists.txt)

--- a/ports/lua/portfile.cmake
+++ b/ports/lua/portfile.cmake
@@ -45,7 +45,7 @@ vcpkg_copy_pdbs()
 vcpkg_cmake_config_fixup(PACKAGE_NAME unofficial-lua CONFIG_PATH share/unofficial-lua)
 
 if("cpp" IN_LIST FEATURES)
-    vcpkg_cmake_config_fixup(PACKAGE_NAME share/unofficial-lua-cpp CONFIG_PATH "share/unofficial-lua-cpp")
+    vcpkg_cmake_config_fixup(PACKAGE_NAME unofficial-lua-cpp CONFIG_PATH "share/unofficial-lua-cpp")
 endif()
 
 if ("tools" IN_LIST FEATURES)

--- a/ports/lua/vcpkg.json
+++ b/ports/lua/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "lua",
   "version": "5.4.4",
+  "port-version": 1,
   "description": "A powerful, fast, lightweight, embeddable scripting language",
   "homepage": "https://www.lua.org",
   "dependencies": [

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -4226,7 +4226,7 @@
     },
     "lua": {
       "baseline": "5.4.4",
-      "port-version": 0
+      "port-version": 1
     },
     "luabridge": {
       "baseline": "2.6",

--- a/versions/l-/lua.json
+++ b/versions/l-/lua.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "26701e6e610ff6eaee198ca1aaa0b9125e4b2d6f",
+      "version": "5.4.4",
+      "port-version": 1
+    },
+    {
       "git-tree": "811a56cbdf5cfd0f8e57c2b3cff2a03e5c709723",
       "version": "5.4.4",
       "port-version": 0

